exercises/list-ops/canonical-data.json

```diff
@@ -222,13 +282,67 @@
         {
           "uuid": "be396a53-c074-4db3-8dd6-f7ed003cce7c",
           "description": "direction dependent function applied to non-empty list",
+          "comments": [
+            "Expects integer division (expects / to discard fractions).       "
+          ],
           "property": "foldr",
           "input": {
             "list": [2, 5],
             "initial": 5,
             "function": "(x, y) -> x / y"
           },
           "expected": 2
+        },
+        {
+          "uuid": "17214edb-20ba-42fc-bda8-000a5ab525b0",
+          "reimplements": "aeb576b9-118e-4a57-a451-db49fac20fdc",
+          "comments": [
+            "Reimplemented to remove ambiguity about the parameters of the    ",
+            "function input."
+          ],
+          "description": "empty list",
+          "property": "foldr",
+          "input": {
+            "list": [],
+            "initial": 2,
+            "function": "(acc, el) -> el * acc"
+          },
+          "expected": 2
+        },
+        {
+          "uuid": "e1c64db7-9253-4a3d-a7c4-5273b9e2a1bd",
+          "reimplements": "c4b64e58-313e-4c47-9c68-7764964efb8e",
+          "comments": [
+            "Reimplemented to remove ambiguity about the parameters of the    ",
+            "function input."
+          ],
+          "description": "direction independent function applied to non-empty list",
+          "property": "foldr",
+          "input": {
+            "list": [1, 2, 3, 4],
+            "initial": 5,
+            "function": "(acc, el) -> el + acc"
+          },
+          "expected": 15
+        },
+        {
+          "uuid": "8066003b-f2ff-437e-9103-66e6df474844",
+          "reimplements": "be396a53-c074-4db3-8dd6-f7ed003cce7c",
+          "comments": [
+            "Reimplemented to remove ambiguity about the parameters of the    ",
+            "function input. Expects / to preserve fractions. Integer division",
+            "will not work here, since it would compute 4 / 24 = 1 / 6. Use   ",
+            "the original test values (be396a53-c074-4db3-8dd6-f7ed003cce7c)  ",
+            "if integer division is expected / required."
+          ],
+          "description": "direction dependent function applied to non-empty list",
+          "property": "foldr",
+          "input": {
+            "list": [1, 2, 3, 4],
+            "initial": 24,
+            "function": "(acc, el) -> el / acc"
```

Comment:
The follow comment concerns **notation** only. It does not change the expected values of any test, or change the semantics of how `foldr` should operate.

I understand that some track maintainers may disagree about whether the function argument list should be **represented as a string** as `(acc, el)` or `(el, acc)`, but I would like to say to those track maintainers that I think we are just going to have to pick one and run with it. Again, I emphasise that either way there would not be a change to the operation of `foldr`: The implementing track should be sure to pass the element as `el` and the accumulator as `acc`, no matter the string representation here.

I advise implementing tracks to be aware of the conventions common to their language do what makes sense for their language. Why do I bring this up? Observe:

* Ramda (a functional library for Javascript): [foldr](https://ramdajs.com/0.9/docs/#foldr) is `(a, b -> a) -> a -> [b] -> a`. The passed function takes its accumulator first and the list element second, and `foldr` takes initial accumulator first and the list second.
* OCaml [List.fold_right](https://caml.inria.fr/pub/docs/manual-ocaml/libref/List.html#VALfold_right) is `('a -> 'b -> 'b) -> 'a list -> 'b -> 'b`. The passed function takes its list element first and the accumulator second, and `fold_right` takes the list first and accumulator second.
* Haskell [Prelude foldr](https://hackage.haskell.org/package/base-4.14.1.0/docs/Prelude.html#v:foldr) is `class Foldable t where foldr :: (a -> b -> b) -> b -> t a -> b`. The passed function takes its list element first and accumulator second, and `foldr` takes the initial accumulator first and the list second.

As you can see, I've listed three different languages doing things in three different ways, so I advise track maintainers looking to squabble over this that there is no universal convention, so we should just pick one for problem-specifications and run with it.

Reply:
This is great advice and I agree 100% with it. I'll link this comment in the comments in the canonical data.